binary_search_tree/tree.py

```diff
@@ -11,45 +11,142 @@ def __init__(self, key, val = None):
 
 
 class Tree:
+    # # ITERATIVE
     def __init__(self):
         self.root = None
 
-    # Time Complexity: 
-    # Space Complexity: 
+    # Time Complexity: O(log n)
+    # Space Complexity: O(1)
     def add(self, key, value = None):
-        pass
+        node = TreeNode(key, value)
+        if not self.root:
+            self.root = node
+            return self.root
+
+        current = self.root
+        previous = None
+        while current:
+            previous = current
+            if key <= current.key:
+                current = current.left
+            else:
+                current = current.right
+        if key <= previous.key:
+            previous.left = node
+        else:
+            previous.right = node
 
+    # RECURSIVE
     # Time Complexity: 
     # Space Complexity: 
+    # def add(self, key, value = None):
+    #     # base case
+    #     if not current.left and not current.right:
```

Comment:
Not a big deal, but generally I recommend removing unfinished but uncommented code like this to just keep the source clean as a style thing.